src/KOKKOS/pair_eam_kokkos.cpp

```diff
@@ -799,6 +843,232 @@ void PairEAMKokkos<DeviceType>::operator()(TagPairEAMKernelC<NEIGHFLAG,NEWTON_PA
   EV_FLOAT ev;
   this->template operator()<NEIGHFLAG,NEWTON_PAIR,EVFLAG>(TagPairEAMKernelC<NEIGHFLAG,NEWTON_PAIR,EVFLAG>(), ii, ev);
 }
+/* ---------------------------------------------------------------------- */
+
+////Specialisation for Neighborlist types Half, HalfThread, Full
+template<class DeviceType>
+template<int EFLAG>
+KOKKOS_INLINE_FUNCTION
+void PairEAMKokkos<DeviceType>::operator()(TagPairEAMKernelAB<EFLAG>,
+		                     const typename Kokkos::TeamPolicy<DeviceType>::member_type& team_member,
+				     EV_FLOAT& ev) const {
+  int ii = team_member.league_rank()*team_member.team_size() + team_member.team_rank();
+  // rho = density at each atom
+  // loop over neighbors of my atoms
+  const int m_max = d_rhor_spline.extent_int(1);
+  const int j_max = d_rhor_spline.extent_int(2);
```

Comment:
iirc this is a compile-time 7

Reply:
IIRC, Ian did test this w/ this hard-coded and we didn't see much of a difference in performance, but I ... think we can change this to a [`static_extent` call](https://github.com/kokkos/kokkos/blob/develop/core/src/Kokkos_View.hpp#L652) such that if the spline degree ever changed (is this fixed for EAM?) we wouldn't miss it there.

Reply:
I think it's fixed for this particular flavor of EAM, it's related to the spline order. You could probably do some type of `static_assert` in the functor to see if the `*7` in `tdual_ffloat_2d_n7`, etc, ever changes as a sanity check?

I'm not surprised it's not making a difference in performance, to a degree this is purely aesthetic... but also, may as well give the compiler a hard-coded number when you can.